tests/installation/system_role/select_role_transactional_server.pm

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Select System Role (Transactional Server) and navigate to next screen
+#          using REST API.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use scheduler 'get_test_suite_data';
+use testapi;
+
+sub run {
+    my $system_role = $testapi::distri->get_system_role_controller();
+
+    $system_role->select_system_role('Transactional Server');
```

Comment:
This [approach](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/24668a3a856f7d61858158edfb3463af9be8a355/lib/Installation/ProductSelection/ProductSelectionPage.pm#L27) has its advantages, I applied because I received feedback about that after first implementation of role selection. Basically we will need to change the tests if the names change or create some sort of additional mapping, it also contains spaces, so if we need for some test to have that as test data (a test that go deeper testing) it would not match our typical nomenclature. 

The idea was to represent all those controls at the level of page, so we just need to use always the same variables of the class, for example 'SLES_for_SAP' to name the variable (this is just to save typing methods) and also in the test (and notice the underscores)  and if the screen changes, the test doesn't change only we overwrite the constructor with the same variables but different search criteria.

Reply:
I found this approach interesting so I wanted to give it a try, but it does not work. The type is different, not YRadiobutton but YSingleItemSelector, and a query like `curl 'http://localhost:9999/v1/widgets?type=YSingleItemSelector&label=Minimal'` returns "widget not found". here is how the json looks like:
[
  {
    "class" : "YSingleItemSelector",
    "hstretch" : true,
    "icon_base_path" : "",
    "id" : "role_selector",
    "items" : 
    [
      {
        "label" : "&Text Mode",
        "selected" : true
      },
      {
        "label" : "&Minimal"
      },
      {
        "label" : "&KVM Virtualization Host"
      },
      {
        "label" : "&XEN Virtualization Host"
      }
    ],
    "items_count" : 4,
    "vstretch" : true
  }
]

also just  `curl 'http://localhost:9999/v1/widgets?label=Minimal'` returns "widget not found" as well.

So basically, it cannot work the same way.

Reply:
ahm, right, in this case is only one control, but applies the same idea, so in order to not change the test if the labels change in future and not use test data, those strings should be at the level of the page, so we can create a method for each selection in SystemRolePage which uses those string (and corresponding methods in controller). If the labels change, we will inherit the page (or replaced if no other product was using it), if they change too often, then we will have to revisit and use test data to favor yaml edition over code edition.